### ISSUE TITLE
How to resume training ? for Yolo-NAS

### ISSUE THREAD

**Issue by AdnanMunir294** [Deci-AI/super-gradients#940]

**Comment by AdnanMunir294** [comment]
As I am training Yolo-Nas my system got stuck to I have ckpt.best_path file. how to resume my training for left epochs? Thanks

**Comment by Louis-Dupont** [comment]
To resume you just need to set `training_params.resume = True`
Feel free to check the [documentation for loading checkpoints](https://docs.deci.ai/super-gradients/documentation/source/Checkpoints.html#loading-checkpoints) if you're interested, it explains more generally how to use checkpoints, with a small section for resuming specifically (`Resuming Training`)

**Comment by absmahi01** [comment]
@Louis-Dupont can you please write the command line code

**Comment by Louis-Dupont** [comment]
Let's say you trained on `training_hyperparams.max_epochs=50` and now want to run for 300 instead:

### If you are training with a [recipe](https://docs.deci.ai/super-gradients/src/super_gradients/recipes/Training_Recipes.html)

```shell
python -m super_gradients.train_from_recipe --config-name=<config-name> experiment_name=<experiment-name> training_hyperparams.resume=True training_hyperparams.max_epochs=300
```
Make sure to keep the same `experiment_name` as in your previous training since it will be used to locate and update the checkpoints.

If you want to create a new experiment based on an existing checkpoint, I invite you to checkout this [documentation page](https://docs.deci.ai/super-gradients/documentation/source/Checkpoints.html#loading-checkpoints)

### If you are running from a script
```python
...

training_params = {...} # Whatever you have been using
training_params['resume'] = True # This will let the Trainer know to load the existing checkpoint and start from the last epoch
training_params['max_epochs'] = 300 # Only if you want to change the number of epochs compared to the previous run

# The following remains the same
trainer.train(model=model, 
              training_params=training_params, 
              train_loader=train_dataloader,
              valid_loader=valid_dataloader)
```


Does this answer your question? @absmahi01 @AdnanMunir294

**Comment by AdnanMunir294** [comment]
but where we will load our ".pth" weights file in script ?
===========================

Mr. Adnan Munir
Graduate Assistant
Dept of Computer Engineering
King Fahd University of Petroleum and Minerals

**Comment by AdnanMunir294** [comment]
[image: image.png]
PLease help I am getting this error
===========================

Mr. Adnan Munir
Graduate Assistant
Dept of Computer Engineering
King Fahd University of Petroleum and Minerals

**Comment by absmahi01** [comment]
@Louis-Dupont but where we will load our ".pth" weights file in script ?

**Comment by AdnanMunir294** [comment]
Done.
train_params['resume'] = True
train_params["resume_path"] =
'/content/checkpoints/my_first_yolonas_run/ckpt_best9.pth'
trainer.train(model=model,
              training_params=train_params,
              train_loader=train_data,
              valid_loader=val_data)
===========================

Mr. Adnan Munir
Graduate Assistant
Dept of Computer Engineering
King Fahd University of Petroleum and Minerals

**Comment by Louis-Dupont** [comment]
@AdnanMunir294 exactly, `resume_path` allows you to specify which checkpoint you want to use to resume your experiment. Note that when using `resume_path` you don't need to set `resume=True` :)

**Comment by Louis-Dupont** [comment]
@absmahi01 , when you set `resume=True`, the model will load the model named `ckpt_latest.pth` in your experiment directory: `<ckpt_root_dir>/<experiment_name>/ckpt_latest.pth`). 
This is usually what you will want to do because you don't need to know the path of your checkpoint, but I see that in your case, you have a checkpoint named `ckpt_best9.pth` (and not `ckpt_latest.pth`), so it makes sense you would want to use `resume_path`

**Comment by AdnanMunir294** [comment]
Thank you for the clarification. Yes my model start training.

**Comment by akashAD98** [comment]
}

#train_params['resume'] = True
train_params["resume_path"] ='checkpoints/base_weapons11_250/ckpt_latest.pth'

im getting new log file & i lost my previous weights, it started training from start, this very bad for me bcz its genrating new logs ,& starting from new weights.


should i need to mentaion resume =true before starting training of model? the last.pt file is not even working

can you please provide more detail doc for resumeing from last .pt & it will update in same log file, insted of making diff log

**Comment by akashAD98** [comment]
@Louis-Dupont size of model im getting is very high? 628 mb, dont you think it will make model slower? or should i need to optimise further ?

**Comment by lsm140** [comment]
> Let's say you trained on `training_hyperparams.max_epochs=50` and now want to run for `300` instead:
> 
> ### If you are training with a [recipe](https://docs.deci.ai/super-gradients/src/super_gradients/recipes/Training_Recipes.html)
> ```shell
> python -m super_gradients.train_from_recipe --config-name=<config-name> experiment_name=<experiment-name> training_hyperparams.resume=True training_hyperparams.max_epochs=300
> ```
> 
> Make sure to keep the same `experiment_name` as in your previous training since it will be used to locate and update the checkpoints.
> 
> If you want to create a new experiment based on an existing checkpoint, I invite you to check out this [documentation page](https://docs.deci.ai/super-gradients/documentation/source/Checkpoints.html#loading-checkpoints)
> 
> ### If you are running from a script
> ```python
> ...
> 
> training_params = {...} # Whatever you have been using
> training_params['resume'] = True # This will let the Trainer know to load the existing checkpoint and start from the last epoch
> training_params['max_epochs'] = 300 # Only if you want to change the number of epochs compared to the previous run
> 
> # The following remains the same
> trainer.train(model=model, 
>               training_params=training_params, 
>               train_loader=train_dataloader,
>               valid_loader=valid_dataloader)
> ```
> 
> Does this answer your question? @absmahi01 @AdnanMunir294

thanks,but during resume training val period，it happened: torch.distributed.elastic.multiprocessing.api][ERROR] - failed (exitcode: -9) how to solve it?or should I modify some parameters?

**Comment by lsm140** [comment]
> Let's say you trained on `training_hyperparams.max_epochs=50` and now want to run for `300` instead:
> 
> ### If you are training with a [recipe](https://docs.deci.ai/super-gradients/src/super_gradients/recipes/Training_Recipes.html)
> ```shell
> python -m super_gradients.train_from_recipe --config-name=<config-name> experiment_name=<experiment-name> training_hyperparams.resume=True training_hyperparams.max_epochs=300
> ```
> 
> Make sure to keep the same `experiment_name` as in your previous training since it will be used to locate and update the checkpoints.
> 
> If you want to create a new experiment based on an existing checkpoint, I invite you to check out this [documentation page](https://docs.deci.ai/super-gradients/documentation/source/Checkpoints.html#loading-checkpoints)
> 
> ### If you are running from a script
> ```python
> ...
> 
> training_params = {...} # Whatever you have been using
> training_params['resume'] = True # This will let the Trainer know to load the existing checkpoint and start from the last epoch
> training_params['max_epochs'] = 300 # Only if you want to change the number of epochs compared to the previous run
> 
> # The following remains the same
> trainer.train(model=model, 
>               training_params=training_params, 
>               train_loader=train_dataloader,
>               valid_loader=valid_dataloader)
> ```
> 
> Does this answer your question? @absmahi01 @AdnanMunir294

Another thing that puzzles me is why my resume training doesn't work when I set resume: True in the yaml file, but it works when I add training_hyperparams.resume=True  on the command line?
